feat: Document the summary of API changes in v3 (#1065)
edgexfoundry/edgex-docs: docs_src/V3TopLevelMigration.md

The Event/Reading data stored by Core Data is considered transient and of little value once it has become old. The V3 versions of these data collections have minimal changes from their V2 counter parts. 

#### API Change

> - In order to identify the device service generating the new event, 
> POST Event REST endpoint will become `/event/{serviceName}/{profileName}/{deviceName}/{sourceName}`

Reviewer: ```suggestion
- Add Event
    POST endpoint is now  `/event/{serviceName}/{profileName}/{deviceName}/{sourceName}`. This is so that the metadata contains the device service which generated the new event
```

Reviewer: Add reference to swagger: https://docs.edgexfoundry.org/{version}/api/core/Ch-APICoreData/

Author: updated

> - Changed database configuration from `Databases map[string]bootstrapConfig.Database` to `Database bootstrapConfig.Database`

Author: updated

Most of the data stored by Core Metadata will be recreated when the V3 versions of the Device Services start-up. The statically declared devices will automatically be created and device discovery will find and add existing devices. Any device profiles, devices, provision watchers created manually via the V2 REST APIs will have to be recreated using the V3 REST API. Any manually-applied `AdministrativeState` settings will also need to be re-applied.

#### API Change
- Remove `LastConnected`, `LastReported` and `UpdateLastConnected` configuration.

> - Changed database configuration from `Databases map[string]bootstrapConfig.Database` to `Database bootstrapConfig.Database`

Author: updated

- Added `optional` field in ResourceProperties
- Updated data type in ResourceProperties
- Updated ProtocolProperties to have typed value
- Updated Value Properties max and min data type from `string` to `number`
- Switched UOM API to YAML format

> - Do not check the empty string profile name when adding the ProvisionWatcher

Reviewer: Be more specific on which APIs these apply to. Add bullet for each API.
Add reference to swagger: https://docs.edgexfoundry.org/{version}/api/core/Ch-APICoreMetadata/

Author: updated

### Core Command
#### API Change

> - Updated `ds-pushevent` and `ds-returnevent` to use bool value

Reviewer: Be more specific on which API, i.e Get Command
Add reference to swagger https://docs.edgexfoundry.org/[version]/api/core/Ch-APICoreCommand/

Author: updated

### Support Notifications

Any `Subscriptions` created via the V2 REST API will have to be recreated using the V3 REST API. The `Notification` and `Transmission`collections will be empty until new notifications are sent using EdgeX V3 

### Support Scheduler
#### API Change

> - Added authentication to support-scheduler actions

Reviewer: Add more details about the new `AuthMethod` field
```suggestion
- Added authentication to support-scheduler actions DTO. 
```

Author: updated

Most of the data stored by Core Metadata will be recreated when the V3 versions of the Device Services start-up. The statically declared devices will automatically be created and device discovery will find and add existing devices. Any device profiles, devices, provision watchers created manually via the V2 REST APIs will have to be recreated using the V3 REST API. Any manually-applied `AdministrativeState` settings will also need to be re-applied.

#### API Change
- Add/ Update/ Get device
    - Remove `LastConnected`, `LastReported` and `UpdateLastConnected` from device model
    - Updated ProtocolProperties to have typed value

- Add/ Update/ Get deviceprofile
    - Added `optional` field in ResourceProperties
    - Updated the data type of `mask`, `shift`, `scale`, `base`, `maximum` and `minimum` from `string` to `number` in ResourceProperties

- Get UOM 
    - Changed the response format from TOML to YAML

- Add/ Update ProvisionWatcher

>     - Allowed empty string profile name when adding or updating the ProvisionWatcher

Reviewer: There are additional changes to the ProvisionWatcher DTO that need to be here. `DiscoveredDevice` device was added, so custom provision watchers need to be migrated to the new DTO.
https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/dtos/provisionwatcher.go#L21

See [Core Metadata API Reference](../../../api/core/Ch-APICoreMetadata.md) for complete details.

### Core Command
#### API Change
- Get Command
    - Updated `ds-pushevent` and `ds-returnevent` to use bool value

See [Core Command API Reference](../../../api/core/Ch-APICoreCommand.md) for complete details.

### Support Notifications

Any `Subscriptions` created via the V2 REST API will have to be recreated using the V3 REST API. The `Notification` and `Transmission`collections will be empty until new notifications are sent using EdgeX V3 

### Support Scheduler
#### API Change

> - Added `authmethod` to support-scheduler actions DTO, which indicates how to authenticate the outbound URL. Its value can be `NONE` or `JWT`.

Author: updated

The Event/Reading data stored by Core Data is considered transient and of little value once it has become old. The V3 versions of these data collections have minimal changes from their V2 counter parts. 

#### API Change
- Add Event

>     POST endpoint is now  `/event/{serviceName}/{profileName}/{deviceName}/{sourceName}`. This is so that the metadata contains the device service which generated the new event

Reviewer 2: > This is so that the metadata contains the device service which generated the new event

is it a correct sentence?  do you understand what it means?

Most of the data stored by Core Metadata will be recreated when the V3 versions of the Device Services start-up. The statically declared devices will automatically be created and device discovery will find and add existing devices. Any device profiles, devices, provision watchers created manually via the V2 REST APIs will have to be recreated using the V3 REST API. Any manually-applied `AdministrativeState` settings will also need to be re-applied.

#### API Change
- Add/ Update/ Get device
    - Remove `LastConnected`, `LastReported` and `UpdateLastConnected` from device model
    - Updated ProtocolProperties to have typed value

- Add/ Update/ Get deviceprofile
    - Added `optional` field in ResourceProperties
    - Updated the data type of `mask`, `shift`, `scale`, `base`, `maximum` and `minimum` from `string` to `number` in ResourceProperties

- Get UOM 
    - Changed the response format from TOML to YAML

- Add/ Get/ Update ProvisionWatcher
    - Allowed empty string profile name when adding or updating the ProvisionWatcher

>     - The ProvisionWatcher DTO is restructured by moving the Device related fields into a new object field, `DiscoveredDevice`; such as `profileName`, `device adminState`, and `autoEvents`.

Reviewer 2: ```suggestion
    - The ProvisionWatcher DTO is restructured by moving the Device related fields into a new object field, `DiscoveredDevice`; such as `profileName`, Device `adminState`, and `autoEvents`.
```

Author: udpated

    - Added a new properties field in the `DiscoveredDevice` object to allow any additional or customized data.

>     - ProvisionWatcher contains its own adminState now. The Device `adminState` is moved into the `DiscoveredDevice` object.

Reviewer 2: ```suggestion
    - ProvisionWatcher contains its own `adminState` now. The Device `adminState` is moved into the `DiscoveredDevice` object.
```

>     - ProvisionWatcher contains its own 1adminState` now. The Device `adminState` is moved into the `DiscoveredDevice` object.

Reviewer 2: ```suggestion
    - ProvisionWatcher contains its own `adminState` now. The Device `adminState` is moved into the `DiscoveredDevice` object.
```

See [Core Metadata API Reference](../../../api/core/Ch-APICoreMetadata.md) for complete details.

### Core Command
#### API Change
- Get Command

>     - Updated `ds-pushevent` and `ds-returnevent` to use bool value

Reviewer 2: ```suggestion
    - Updated `ds-pushevent` and `ds-returnevent` to use bool value, `true` or `false`, instead of `yes` or `no`
```

Author: updated

See [Core Command API Reference](../../../api/core/Ch-APICoreCommand.md) for complete details.

### Support Notifications

Any `Subscriptions` created via the V2 REST API will have to be recreated using the V3 REST API. The `Notification` and `Transmission`collections will be empty until new notifications are sent using EdgeX V3 

### Support Scheduler
#### API Change
- Added `authmethod` to support-scheduler actions DTO, which indicates how to authenticate the outbound URL. Use `NONE` when running in non-secure mode and `JWT` when running in secure mode.

See [Support Scheduler API Reference](../../../api/support/Ch-APISupportScheduler.md) for complete details.

> See [Support Scheduler API Reference](../../../api/support/Ch-APISupportScheduler.md) for complete details.

Reviewer 2: duplicated sentence

Author: updated
